reporters/reporter-default/src/main/java/org/radargun/reporting/html/ReportDocument.java

```diff
@@ -266,110 +183,46 @@ private double toMillis(double nanos) {
-         presentedStatistics.add(StatisticType.OPERATION_THROUGHPUT);
-      }
-      if (hasRepresentation(operation, reportAggregationMap, DataThroughput.class, 100L, 100L, 100L, 100.0)) {
-         presentedStatistics.add(StatisticType.DATA_THROUGHPUT);
-      }
+   //helper class for FreeMarker template
+   public class OperationData {
+      private List<String> iterationValues;
+      private Collection<StatisticType> presentedStatistics;
 
-      List<String> iterations = new ArrayList<>(maxIterations);
-      for (int iteration = 0; iteration < maxIterations; ++iteration) {
-         // in fact we shouldn't have different iterations values for iterations with the same id,
-         // but it's possible
-         Set<String> iterationValues = new HashSet<>();
-         for (List<Aggregation> aggregations : reportAggregationMap.values()) {
-            if (aggregations != null && iteration < aggregations.size()) {
-               Aggregation aggregation = aggregations.get(iteration);
-               if (aggregation != null && aggregation.iteration.getValue() != null) {
-                  iterationValues.add(aggregation.iteration.test.iterationsName + " = " + aggregation.iteration.getValue());
-               }
-            }
-         }
-         iterations.add(concatOrDefault(iterationValues, "iteration " + String.valueOf(iteration)));
+      public OperationData(Collection<StatisticType> presentedStatistics, List<String> iterationValues) {
+         this.presentedStatistics = presentedStatistics;
+         this.iterationValues = iterationValues;
       }
 
-      writeOperationHeader(iterations, presentedStatistics, singleTestName);
-      for (Map.Entry<Report, List<Aggregation>> entry : reportAggregationMap.entrySet()) {
-         writeOperationLine(operation, presentedStatistics, entry.getKey(), entry.getValue());
+      public List<String> getIterationValues() {
+         return iterationValues;
       }
-      write("</table><br>\n");
 
-      for (Future f : chartTaskFutures) {
-         try {
-            f.get();
-         } catch (Exception e) {
-            log.error("Failed to create chart", e);
-         }
+      public Collection<StatisticType> getPresentedStatistics() {
+         return presentedStatistics;
       }
-      chartTaskFutures.clear();
    }
 
-   private void writeOperationLine(String operation, Collection<StatisticType> presentedStatistics, Report report, List<Aggregation> aggregations) {
-      int nodeCount = aggregations.isEmpty() ? 0 : aggregations.get(0).nodeStats.size();
-      int expandableRows = 0;
-      if (configuration.generateNodeStats) {
-         expandableRows += nodeCount;
-         if (configuration.generateThreadStats) {
-            for (int node = 0; node < nodeCount; ++node) {
-               expandableRows += getMaxThreads(aggregations, node);
-            }
-         }
-      }
+   public Class defaultOutcomeClass() {
+      return DefaultOutcome.class;
```

Comment:
not used. You should not create method for this or you need value for injecting into template?